lib/ansible/modules/files/copy.py

```diff
@@ -612,7 +614,7 @@ def main():
         if os.access(b_dest, os.R_OK) and os.path.isfile(b_dest):
             checksum_dest = module.sha1(dest)
     else:
-        if not os.path.exists(os.path.dirname(b_dest)):
+        if not os.path.exists(os.path.dirname(b_dest)) and not module.check_mode:
```

Comment:
I think a more comprehensive solution is needed here in check mode. If the directory already exists, the permissions check should still be performed. If it does not exist, check the permissions on the nearest parent directory that does exist.

This assumes the module will have write access to any additional directories it creates, compared to the existing behavior where that is confirmed by actually creating missing directories.

Reply:
> This assumes the module will have write access to any additional directories it creates, compared to the existing behavior where that is confirmed by actually creating missing directories.

That makes sense. I will refactor this to check that.